test-cases/tests/authorization/b09-validate-authentication-with-ldap-idp.md

```diff
@@ -0,0 +1,117 @@
+
+We want to validate that RHOAM is able to authenticate to an LDAP server via IDP
+
+## Steps
+
+First lets start the AWS EC2 instance that have the ldap service installed
+
+1. Access the AWS EC2 console, you can find the credentials in https://gitlab.cee.redhat.com/integreatly-qe/vault/-/blob/master/SECRETS.md
+
+2. On the AWS EC2 console, right click on the LDAP Server EC2 instance and select start instance.
+
+   > Instance should change to running state
+
+3. Once the instance is in a running state verify if the LDAP service is up and running.
+
+   ```bash
+   // you need to change the <EC2 instance IP> to the EC2 instance IP
+
+   curl -vv "ldap://<EC2 instance IP>:389/dc=ec2-3-133-150-27,dc=us-east-2,dc=compute,dc=amazonaws,dc=com?uid?sub?(uid=rhoam)"
+
+   DN: uid=rhoam,ou=people,dc=ec2-3-133-150-27,dc=us-east-2,dc=compute,dc=amazonaws,dc=com
+    uid: rhoam
+   ```
+
+Now that the LDAP Server is up and running we need to create the integration with RHOAM via IDP.
+
+1. On the OCM console https://qaprodauth.cloud.redhat.com/openshift, go to the cluster you want to add the IDP.
+
+2. Click on the `Access Control` tab.
+
+3. Press the `Add identity provider` button.
+
+4. A form will pop up in a modal with three sections.
+
+### Step 1: Select identity providers type
+
+1. Select the LDAP option in the `Indentity provider` field
+
+### Step 2: Enter provider type information
+
+1. Leave the fields `Name` and `Mapping method` as it is
+
+2. In `LDAP URL` add the URL we want to use for searching users in the LDAP server
+
+   ```bash
+   // you need to change the <EC2 instance IP> to the EC2 instance IP
+
+   ldap://<EC2 instance IP>:389/dc=ec2-3-133-150-27,dc=us-east-2,dc=compute,dc=amazonaws,dc=com?uid?sub
+   ```
+
+3. Leave the fields Bind DN and Bind password empty
+
+### Attributes
+
+1. In ID field add `dn`
+
+2. In `Preferred username` enter `uid`
+
+3. In `Name` enter `cn`
+
+4. Click on `Show Advanced Options` and mark the `Insecure` checkbox, so we don't need to configure certs
+
+5. Press the `Confirm` button and wait for the changes to be reconciled
+
+Lets verify if the integration worked as expected
+
+1. Go to the cluster login page and check if a LDAP option for authentication shows up, if so click on this option
+
+2. Enter _rhoam_ for `username` and `password` and click the log in button
+
+   > You should be redirected to the cluster console main page
+
+3. In the Cluster web console go to, redhat-rhoam-3scale namespace, and then select `Networking >> Routes`.
+
+4. Under the Location column copy the URL that matches with `https://3scale-admin.apps.<cluster-id>.devshift.org`, open an incognito window and paste the url in.
+
+   > You should be redirected to the 3scale login page
+
+5. Click on the `Authenticate through Red Hat Single Sign-On` link
+
+   > You should be redirected to Red Hat single sign on page
+
+6. Click on the LDAP option.
+
+7. Enter _rhoam_ for `username` and `password` and click the log in button.
```

Comment:
do you think it would make sense to have 2 users, that we would use for `customer-admin-user` and `regular-user` scenarios? e.g. `rhoam-test-user` and `rhoam-customer-admin`?

Reply:
hmm, not sure it would make much difference, all users when they are imported they will be created as `test-user` afaik unless we can add some flag in the ldap server that the operator can detect and distinguish which group they belong. 

When I was testing I manually changed the user between `dedicated-admin` and `test-user` and it worked as expected.

Reply:
We're going to perform a full test run against a cluster with LDAP IDP (i.e. all relevant manual tests from test case folder). Some tests require logging in as regular user, some require customer-admin permissions, so I think having 2 users with dedicated permissions could be helpful for us.

Also I think that adding a step for changing the permissions for the customer(dedicated) admin in ocm should be sufficient.

Wdyt?

Reply:
in this case that makes sense to have two users, one dedicated-admin and regular user, I'll create them a update the test case